core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -81,6 +81,26 @@ public static TableMetadata newTableMetadata(
         schema, spec, sortOrder, location, unreservedProperties(properties), formatVersion);
   }
 
+  public static TableMetadata newTableMetadataWithDefaultProperties(
```

Comment:
The name kind of indicates that all defaults are persisted. Are there any better alternatives?

Reply:
After thinking about this a bit more, I'd probably just update the existing method given that we modify all places we know and that's the behavior we want to achieve.

Reply:
Alright then , just change the existing method, hope there's no breakages